### PR TITLE
Add comment and handle intersection types in gotoType

### PR DESCRIPTION
**Pull request by mhegazy**
Follow up on #4632

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -3994,24 +3994,24 @@ namespace ts {
 
             if (flags & SymbolFlags.Property) {
                 if (flags & SymbolFlags.SyntheticProperty) {
-                    // If union property is result of union of non method (property/accessors/variables), it is labeled as property
-                    let unionPropertyKind = forEach(typeChecker.getRootSymbols(symbol), rootSymbol => {
+                    // If synthetic property is result of union of non method (property/accessors/variables), it is labeled as property
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: synthetic -> a synthetic

union of -> union/intersection of

File: src/services/services.ts [hunk 2]
```diff
@@ -3994,24 +3994,24 @@ namespace ts {
...
                             return ScriptElementKind.memberVariableElement;
                         }
                         Debug.assert(!!(rootSymbolFlags & SymbolFlags.Method));
                     });
-                    if (!unionPropertyKind) {
-                        // If this was union of all methods,
-                        //make sure it has call signatures before we can label it as method
-                        let typeOfUnionProperty = typeChecker.getTypeOfSymbolAtLocation(symbol, location);
-                        if (typeOfUnionProperty.getCallSignatures().length) {
+                    if (!syntheticPropertyKind) {
```
Comment (vladima) [inline src/services/services.ts]: I would rewrite it as
```ts
if (syntheticPropertyKind) {
    return syntheticPropertyKind;
}
let typeOfSyntheticProperty = typeChecker.getTypeOfSymbolAtLocation(symbol, location);
return typeOfSyntheticProperty.getCallSignatures().length 
    ?  ScriptElementKind.memberFunctionElement
    : ScriptElementKind.memberVariableElement;
```

Comment (vladima) [inline src/services/services.ts]: btw, are all changes in this block just renames?

Comment (mhegazy) [inline src/services/services.ts]: yes.

**Comment by DanielRosenwasser** [comment]
Can you port this into `master` @mhegazy?

**Comment by DanielRosenwasser** [comment]
Ping

**Comment by DanielRosenwasser** [comment]
Ping

**Comment by RyanCavanaugh** [comment]
He's in Egypt :wink: Does someone else need to handle this? I don't see a bug on it

**Comment by mhegazy** [comment]
this is fairly stale by now. closing.

**Pull request closed without merging**